langchain/guardrails/format_instructions.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from typing import Any
+
+from langchain.chains.llm import LLMChain
+from langchain.output_parsers.base import BaseOutputParser, OutputParserException
+from langchain.prompts.base import PromptValue
+from langchain.prompts.prompt import PromptTemplate
+from langchain.schema import BaseLanguageModel, GuardedOutputParser
+
+TEMPLATE = """Prompt:
+{prompt}
+Completion:
+{completion}
+
+Above, the Completion did not satisfy the constraints given in the Prompt.
+Please try again:"""
+
+
+PROMPT = PromptTemplate.from_template(TEMPLATE)
+
+
+class FormatInstructionsGuard(GuardedOutputParser):
```

Comment:
the names are pretty confusing (a Guard can not be an instance of a Parser?), but think we're going in circles with this. that said, it's not guardrailing the instructions, it's guardrailing the parsing? 

what if this PR just introduces one non-abstract class GuardedOutputParser. No need for class hierarchy right now.